src/Psl/Math/clamp.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Math;
+
+use Psl;
+
+/**
+ * Returns a number whose value is limited to the given range.
+ *
+ * @template T of float|int
+ *
+ * @param T $number
+ * @param T $min
+ * @param T $max
+ *
+ * @return T
+ * @throws Psl\Exception\InvariantViolationException If min is bigger than max
+ *
+ * @psalm-pure
+ */
+function clamp($number, $min, $max)
+{
+    Psl\invariant($min <= $max, 'Expected %s to be lower or equal to %s.', $min, $max);
```

Comment:
we should avoid having parameters added to the exception message ( as they can be dynamic and you don't really want dynamic messages in logs as it would be hard to group them together ) 

ref: https://github.com/azjezz/psl/blob/1.0.x/src/Psl/Math/base_convert.php#L47

```suggestion
    Psl\invariant($min <= $max, 'Expected $min to be lower or equal to $max.');
```